students/biegon_piotr/lesson_03_functions/input_validation.py

```diff
@@ -0,0 +1,21 @@
+print("Input Validation\n")
+
+
+def collatz(number):
+    if number % 2 == 0:
+        return number // 2
+    else:
+        return 3 * number + 1
+
+
+while 1:
+    try:
+        number = int(input("Enter a number: "))
+        if collatz(number) != 1:
```

Comment:
in `try...except` block we should not put much code. the less the better. Would `print(collatz(number))` raise an ValueError exception ?

Reply:
Take a look at  `try...except...elese` clause.
From https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python:

> Without the else-clause, the only option to run additional code before finalization would be the **clumsy practice of adding the code to the try-clause**. That is clumsy because it risks raising exceptions in code that wasn't intended to be protected by the try-block.